### PR TITLE
Imaginary friend fixes

### PR DESCRIPTION
**Pull request by DominikPanic**
The message should only display once now, they should be less hacky.

### CODE REVIEW

File: code/modules/admin/fun_verbs.dm [hunk 1]
```diff
@@ -977,40 +977,26 @@
...
-				return
-			L = C.mob
-		if("Mob")
-			var/mob/M = input("Please, select a mob.", "Imaginary Friend") as null|anything in sortNames(GLOB.mob_living_list)
-			if(!M)
-				return
-			L = M
-
-	if(!isobserver(usr))
+	if(is_mentor(C) && !isobserver(C.mob))
```
Comment (Rohesie) [inline code/modules/admin/fun_verbs.dm]:
```
if(!isobserver(C.mob))
	if(is_mentor(C)
		usr.client.holder.admin_ghost()
		to_chat(C, "<span class='warning'>Can only become an imaginary friend while observing.</span>")
		return
	C.holder.admin_ghost()
```

Comment (DominikPanic) [reply code/modules/admin/fun_verbs.dm]: That is changing the functionality to something completely different

Comment (Rohesie) [reply code/modules/admin/fun_verbs.dm]: Unless I'm missing something the behavior should remain exactly the same this way.

Comment (DominikPanic) [reply code/modules/admin/fun_verbs.dm]: I want to return early for mentors, not admin ghost them, i want to only admnghost admins.

Comment (Rohesie) [reply code/modules/admin/fun_verbs.dm]: That's what will happen with my code. Non-observer mentors will trigger `if(!isobserver(C.mob))` and `if(is_mentor(C))`, reaching the `return` before the `C.holder.admin_ghost()`.

Comment (Rohesie) [reply code/modules/admin/fun_verbs.dm]: Though it would be more maintainable if it was something like `!is_admin` instead.

Comment (DominikPanic) [reply code/modules/admin/fun_verbs.dm]:
```
	if(is_mentor(C))
		usr.client.holder.admin_ghost()
```
No it wont

Comment (Rohesie) [reply code/modules/admin/fun_verbs.dm]: Heck, I only saw that now. Very much unintended. Remove that line and the rest follows.

**Pull request merged**